fsproxy/src/main/java/no/unit/identityservice/fsproxy/FsApi.java

```diff
@@ -0,0 +1,189 @@
+package no.unit.identityservice.fsproxy;
+
+import static nva.commons.core.attempt.Try.attempt;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.attribute.UserPrincipalNotFoundException;
+import java.time.Year;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import no.unit.identityservice.fsproxy.model.course.FsCourse;
+import no.unit.identityservice.fsproxy.model.course.FsCoursesSearchResult;
+import no.unit.identityservice.fsproxy.model.person.FsIdNumber;
+import no.unit.identityservice.fsproxy.model.person.FsPersonSearchResponse;
+import no.unit.identityservice.fsproxy.model.person.NationalIdentityNumber;
+import no.unit.identityservice.fsproxy.model.staffperson.FsCourseActivity;
+import no.unit.identityservice.fsproxy.model.staffperson.FsRoleToStaffPerson;
+import no.unit.identityservice.fsproxy.model.staffperson.FsRolesToPersonSearchResult;
+import no.unit.identityservice.fsproxy.model.staffperson.FsUriToCourseActivity;
+import no.unit.identityservice.fsproxy.model.staffperson.FsUriToCourseActivityContainer;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.attempt.Try;
+import nva.commons.core.paths.UriWrapper;
+
+public class FsApi {
+
+    public static final String BIRTHDATE_IDENTIFIER = "fodselsdato0";
+    public static final String PERSONAL_NUMBER_IDENTIFIER = "personnummer0";
+    public static final String PERSON_ROLES = "personroller";
+    public static final String SEMESTER_YEAR_PATH = "semester.ar";
+    public static final String PERSON_PATH = "personer";
+    public static final String STUDENT_TEACHING_PATH = "studentundervisning";
+    public static final String TEACHING_SEMESTER_AR_PATH = "undervisning.semester.ar";
+    public static final String FS_PERSON_ID_NUMBER_PATH = "person.personlopenummer";
+    public static final String DB_IDENTIFIER = "dbId";
+    public static final String LIMIT_IDENTIFIER = "limit";
+    public static final String LIMIT_VALUE = "0";
+    private static final Environment ENVIRONMENT = new Environment();
+    public static final URI FS_HOST = readFsHost();
+    private final String username = ENVIRONMENT.readEnv("FS_USERNAME");
+    private final String password = ENVIRONMENT.readEnv("FS_PASSWORD");
+    private final URI baseFsHostUrl;
+    private final HttpClient httpClient;
+
```

Comment:
No custom log messages required?
I am just asking. If the exceptions messages are good then we don't need more spam in the logs.

Reply:
har lagt inn en logger